docs/src/Button.doc.js

```diff
@@ -9,281 +9,287 @@ import PageHeader from './components/PageHeader.js';
-        href: 'accessibility-disclosure',
+        required: false,
+        defaultValue: null,
+        description: buildDescription([
+          'Indicate that a button component hides or exposes collapsible components and expose whether they are currently expanded or collapsed.',
+          'Optional with button-type buttons.',
+          'Accessibility: It populates aria-expanded.',
+        ]),
+        href: 'accessibilityII',
       },
       {
         name: 'accessibilityHaspopup',
         type: 'boolean',
-        description:
-          'Indicates that the element has a popup context menu or sub-level menu. Populates aria-haspopup.',
-        href: 'accessibility-popup',
-      },
-      {
-        name: 'accessibilityLabel',
-        type: 'string',
-        description:
-          'String that clients such as VoiceOver will read to describe the element. Populates aria-label.',
-        href: 'accessibility-popup',
+        required: false,
+        defaultValue: null,
+        description: buildDescription([
+          'Indicate that a button component controls the appearance of menus and expose whether they are currently opened or closed.',
+          'Optional with button-type buttons.',
+          'Accessibility: It populates aria-haspopup.',
+        ]),
+        href: 'accessibilityII',
       },
       {
         name: 'color',
-        type: `"blue" | "gray" | "red" | "transparent" | "white"`,
+        type: `gray' | 'red' | 'blue' | 'transparent' | 'white'`,
+        required: false,
         defaultValue: 'gray',
-        href: 'color',
-      },
-      {
-        name: 'disabled',
-        type: 'boolean',
-        defaultValue: false,
+        description: 'Primary colors to apply to the button background.',
         href: 'combinations',
       },
       {
-        name: 'selected',
+        name: 'disabled',
         type: 'boolean',
-        defaultValue: false,
+        required: false,
+        defaultValue: null,
+        description:
+          'Set disabled state so buttons look inactive, cannot be interacted with, and actions are not available.',
         href: 'combinations',
       },
       {
         name: 'iconEnd',
         type: '$Keys<typeof icons>',
-        description:
-          'Add an icon to be displayed after the button text. This allows type checking for a valid icon name based on the keys from the list of icons in Icon.',
+        required: false,
+        defaultValue: false,
+        description: buildDescription([
+          'Add a Gestalt icon to be displayed after the button text. Sometimes an icon can help clarify the usage of a button. Menus are a common use case.',
+          'Accessibility: Icons on buttons are not accessible for screen readers.',
+          'Use IconButton if you only need buttons with icons and no text.',
+        ]),
         href: 'iconEnd',
       },
       {
         name: 'inline',
         type: 'boolean',
+        required: false,
```

Comment:
@AlbertCarreras curious why we added `required: false`, is that not the default?